Src/Newtonsoft.Json.UnityConverters/VectorConverter.cs

```diff
@@ -0,0 +1,184 @@
+            EnableVector4 = true;
+        }
+
+        /// <summary>
+        /// Selectively enable Vector types
+        /// </summary>
+        /// <param name="enableVector2">Use for Vector2 objects</param>
+        /// <param name="enableVector3">Use for Vector3 objects</param>
+        /// <param name="enableVector4">Use for Vector4 objects</param>
+        public VectorConverter(bool enableVector2, bool enableVector3, bool enableVector4): this()
+        {
+            EnableVector2 = enableVector2;
+            EnableVector3 = enableVector3;
+            EnableVector4 = enableVector4;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="writer"></param>
+        /// <param name="value"></param>
+        /// <param name="serializer"></param>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            if (value == null)
+            {
+                writer.WriteNull();
+                return;
+            }
+
+            var targetType = value.GetType();
+
+            if (targetType == V2)
+            {
+                var targetVal = (Vector2)value;
+                WriteVector(writer, targetVal.x, targetVal.y, null, null);
+            }
+            else if (targetType == V3)
+            {
+                var targetVal = (Vector3) value;
+                WriteVector(writer, targetVal.x, targetVal.y, targetVal.z, null);
+            }
+            else if (targetType == V4)
+            {
+                var targetVal = (Vector4)value;
+                WriteVector(writer, targetVal.x, targetVal.y, targetVal.z, targetVal.w);
+            }
+            else
+            {
+                //Should never get here
+                writer.WriteNull();
+            }
+
+        }
+
+        private static void WriteVector(JsonWriter writer, float x, float y, float? z, float? w)
+        {
+            writer.WriteStartObject();
+
+            writer.WritePropertyName("x");
+            writer.WriteValue(x);
+            writer.WritePropertyName("y");
+            writer.WriteValue(y);
+
+            if (z.HasValue)
+            {
+                writer.WritePropertyName("z");
+                writer.WriteValue(z.Value);
+
+                if (w.HasValue)
+                {
+                    writer.WritePropertyName("w");
+                    writer.WriteValue(w.Value);
+                }
+            }
+
+            writer.WriteEndObject();
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="reader"></param>
+        /// <param name="objectType"></param>
+        /// <param name="existingValue"></param>
+        /// <param name="serializer"></param>
+        /// <returns></returns>
+        public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
+        {
+            if (objectType == V2)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/jilleJr/Newtonsoft.Json-for-Unity.Converters/pullRequest?prid=4846771)